src/panel-toc-editor.ts

```diff
@@ -0,0 +1,259 @@
+        children: orphanModules.map(moduleObjectFromModuleId).sort((m, n) => moduleIdNumber(m.moduleid) - moduleIdNumber(n.moduleid))
+      };
+
+      messageQueued = {
+        uneditable: [collectionAllModules, collectionOrphanModules],
+        editable: collectionTrees
+      }
+    }
+  
+    panel.webview.onDidReceiveMessage(async (message) => {
+      const { signal } = message;
+      if (signal != null) {
+        if (signal === 'loaded') {
+          panel.webview.postMessage(messageQueued)
+        }
+      }
+      const { treeData } = message;
+      const uri = getRootPathUri();
+      if (uri != null && treeData != null) {
+        const replacingUri = uri.with({ path: path.join(uri.fsPath, 'collections', `${treeData.slug}.collection.xml`)});
+        const collectionData = fs.readFileSync(replacingUri.fsPath, { encoding: 'utf-8' });
+        const document = new DOMParser().parseFromString(collectionData);
+        replaceCollectionContent(document, treeData);
+        const serailizedXml = xmlFormat(new XMLSerializer().serializeToString(document), {
+          indentation: '  ',
+          collapseContent: true,
+          lineSeparator: '\n'
+        });
+        console.log(`writing: ${replacingUri.toString()}`);
+        const textDocument = await vscode.workspace.openTextDocument(replacingUri);
+        const fullRange = new vscode.Range(
+          textDocument.positionAt(0),
+          textDocument.positionAt(textDocument.getText().length)
+        );
+        const edit = new vscode.WorkspaceEdit();
+        edit.replace(replacingUri, fullRange, serailizedXml);
+        vscode.workspace.applyEdit(edit);
+      }
+    });
+  };
+}
+
+function moduleIdNumber(moduleid: string): number {
+	const numberPart = moduleid.substr(1);
+	return parseInt(numberPart);
+}
+
+function insertUsedModules(arr: Array<string>, tree: any){
+	if (tree.type === 'module') {
+		arr.push(tree.moduleid);
+	} else if (tree.children) {
+		for (const child of tree.children) {
+			insertUsedModules(arr, child);
+		}
+	}
+}
+
+function moduleObjectFromModuleId(moduleid: string): any {
+	return {
+    type: 'module',
+		moduleid: moduleid,
+		title: getModuleTitle(moduleid)
+  };
+}
+
+function moduleToObject(element: any): any {
+	const moduleid = element.getAttribute('document');
+	if (moduleid == null) {
+		throw new Error('Error parsing collection. ModuleID missing.');
+	}
+  return moduleObjectFromModuleId(moduleid);
+}
+
+function subcollectionToObject(element: any): any {
+  const title = element.getElementsByTagNameNS(NS_METADATA, 'title')[0].textContent;
+	const content = element.getElementsByTagNameNS(NS_COLLECTION, 'content')[0];
+	if (title == null) {
+		throw new Error('Error parsing collection. Subcollection title missing.');
+	}
+  return {
+    type: 'subcollection',
+    title: title,
+    children: childObjects(content)
+  };
+}
+
+function childObjects(element: any): Array<any> {
+	const children = [];
```

Comment:
There seems to be inconsistent indentation; is this not picked up by the linter?

Reply:
ugh someone there are some tabs and some spaces :vomiting_face:

Reply:
for some reason, the linter doesn't seem to care...

Reply:
I will address while switching out the linter in a different pr

Reply:
card created